test/python/circuit/test_controlled_gate.py

```diff
@@ -0,0 +1,402 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=unused-import
+
+"""Test Qiskit's inverse gate operation."""
+
+import os
+import tempfile
+import unittest
+from math import pi
+import numpy as np
+from ddt import ddt, data
+
+from qiskit import (QuantumRegister, ClassicalRegister, QuantumCircuit, execute,
+                    BasicAer)
+from qiskit.test import QiskitTestCase
+from qiskit.circuit import ControlledGate
+from qiskit.compiler import transpile
+from qiskit.converters.instruction_to_gate import instruction_to_gate
+from qiskit.extensions.standard import CnotGate
+from qiskit.quantum_info.operators.predicates import matrix_equal, is_unitary_matrix
+import qiskit.circuit.add_control as ac
+from qiskit.transpiler.passes import Unroller
+from qiskit.converters.circuit_to_dag import circuit_to_dag
+from qiskit.converters.dag_to_circuit import dag_to_circuit
+from qiskit.quantum_info import Operator
+from qiskit.extensions.standard import (CnotGate, XGate, YGate, ZGate, U1Gate,
+                                        CyGate, CzGate, Cu1Gate, SwapGate)
+from qiskit.extensions.unitary import UnitaryGate
+
+
+@ddt
+class TestControlledGate(QiskitTestCase):
+    """ControlledGate tests."""
+
+    def test_controlled_x(self):
+        """Test creation of controlled x gate"""
+        self.assertEqual(XGate().q_if(), CnotGate())
+
+    def test_controlled_y(self):
+        """Test creation of controlled x gate"""
+        self.assertEqual(YGate().q_if(), CyGate())
+
+    def test_controlled_z(self):
+        """Test creation of controlled x gate"""
+        self.assertEqual(ZGate().q_if(), CzGate())
+
+    def test_controlled_u1(self):
+        """Test creation of controlled x gate"""
+        theta = 0.5
+        self.assertEqual(U1Gate(theta).q_if(), Cu1Gate(theta))
+
+    def test_circuit_append(self):
+        """Test appending controlled gate to quantum circuit"""
+        circ = QuantumCircuit(5)
+        inst = CnotGate()
+        circ.append(inst.q_if(), qargs=[0, 2, 1])
+        circ.append(inst.q_if(2), qargs=[0, 3, 1, 2])
+        circ.append(inst.q_if().q_if(), qargs=[0, 3, 1, 2])  # should be same as above
```

Comment:
can you add a test to test this "should be same as above"?
There are unit tests for x->cx, y->cy, z->cz, u1->cu1... 
can you add unit tests to cover
cx->ccx, u3->cu3, h->ch, swap->cswap?

Reply:
done